bigcache.go

```diff
@@ -33,6 +33,79 @@ type cacheShard struct {
 	onRemove    func(wrappedEntry []byte)
 }
 
+// EntryInfo holds informations about entry in the cache
+type EntryInfo struct {
+	Key       string
+	Hash      uint64
+	Timestamp uint64
+	index     uint32
+}
+
+// EntryInfoIterator allows to iterate over entries in the cache
+type EntryInfoIterator struct {
+	cache        *BigCache
+	currentShard int
+	currentIndex int32
+	elements     []*EntryInfo
+}
+
+func copyCurrentShardMap(shard *cacheShard) []*EntryInfo {
+	shard.lock.RLock()
+	defer shard.lock.RUnlock()
+
+	elements := make([]*EntryInfo, 0)
```

Comment:
Try to estimate initial size for elements as `len(shard.queue)/cache.config.MaxEntrySize`